app/models/name/merge.rb

```diff
@@ -130,4 +134,40 @@ def merge(old_name)
     # Finally destroy the name.
     old_name.destroy
   end
+
+  ##############################################################################
+
+  private
+
+  def approved_synonym_of_proposed_name?
+    !deprecated && (other_synonym_ids & Naming.all.pluck(:name_id)).present?
```

Comment:
I worry about how this is implemented under the hood.  Have you checked?  It could be returning the entire list of name_ids of every naming on MO(!) then intersecting that with other_synonym_ids.  It would be better to let mysql do the work and just return a "yay" or "nay".  Would this work?

Naming.where(name_id: other_synonym_ids).any?

Reply:
Another good question. Thanks for continually prodding me on performance.
I'll re-examine this.

Reply:
> Another good question. Thanks for continually prodding me on performance.
> I'll re-examine this.

`(other_synonym_ids & Naming.all.pluck(:name_id)).present?` is awful in exactly the way you politely suggest. I'll fix it along the lines of your suggested ActiveRecord query.